lib/jekyll/drops/drop.rb

```diff
@@ -199,6 +199,10 @@ def merge!(other)
           end
         end
       end
+
+      def fetch(*args, &block)
+        to_h.fetch(*args, &block)
```

Comment:
`#to_h` is an expensive operation. Could we mimic the behaviour without having to actually create a hash?